openhtf/io/output/json_factory.py

```diff
@@ -44,19 +44,12 @@ def default(self, obj):
   def __call__(self, test_record):
     assert self.filename_pattern, 'filename_pattern required'
     if self.inline_attachments:
-      as_dict = data.ConvertToBaseTypes(test_record)
-      for phase in as_dict['phases']:
+      self.test_record_dict = data.ConvertToBaseTypes(test_record)
```

Comment:
generally avoid putting things on `self` when you don't need to, also, any attributes you plan to set on `self` should be initialized in `__init__` (this is a matter of style/convention, it's not technically required by Python).  In this case, you shouldn't save it on `self`.